src/StakingNodesManager.sol

```diff
@@ -441,6 +464,40 @@ contract StakingNodesManager is
         emit WithdrawnETHRewardsProcessed(nodeId, rewardsType, msg.value);
     }
 
+    /// @notice Processes principal withdrawals, specifying how much goes back into ynETH and how much goes to the withdrawal queue.
+    /// @param nodeId The ID of the node processing the withdrawal.
+    /// @param amountToReinvest Amount of ETH to reinvest into ynETH.
+    /// @param amountToQueue Amount of ETH to send to the withdrawal queue.
+    function processPrincipalWithdrawalsForNode(
```

Comment:
this permissioned potentially can be improved.

Gives full control currently for allocating funds back into pool vs withdrawal